app-modules/report/src/Filament/Widgets/PromptsByCategoryDoughnutChart.php

```diff
@@ -0,0 +1,106 @@
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Report\Filament\Widgets;
+
+use Filament\Widgets\ChartWidget;
+use AdvisingApp\Ai\Models\PromptType;
+
+class PromptsByCategoryDoughnutChart extends ChartWidget
+{
+    protected static ?string $heading = 'Prompts by Category';
+
+    protected static ?string $pollingInterval = null;
+
+    protected int | string | array $columnSpan = [
+        'sm' => 1,
+        'md' => 1,
+        'lg' => 1,
+    ];
+
+    protected function getOptions(): array
+    {
+        return [
+            'maintainAspectRatio' => false,
+            'plugins' => [
+                'legend' => [
+                    'display' => true,
+                ],
+            ],
+            'scales' => [
+                'x' => [
+                    'display' => false,
+                ],
+                'y' => [
+                    'display' => false,
+                ],
+            ],
+        ];
+    }
+
+    protected function getData(): array
+    {
+        $promptsByCategory = PromptType::withCount(['prompts'])->get(['id', 'title']);
+
+        $promptsByCategory = $promptsByCategory->map(function (PromptType $promptType) {
+            $promptType['bg_color'] = $this->getRgbString();
+
+            return $promptType;
+        });
+
+        return [
+            'labels' => $promptsByCategory->pluck('title'),
+            'datasets' => [
+                [
+                    'label' => 'My First Dataset',
+                    'data' => $promptsByCategory->pluck('prompts_count'),
+                    'backgroundColor' => $promptsByCategory->pluck('bg_color'),
+                    'hoverOffset' => 4,
+                ],
+            ],
+        ];
```

Comment:
We should cache this value for something like 15 minutes: https://laravel.com/docs/11.x/cache#retrieve-store

And clear the cache key when this value would be affected elsewhere in the app. Such as when a PromptType is created/deleted or when a Prompt is created or updated to change the PromptType